matchers/matchers.go

```diff
@@ -0,0 +1,156 @@
+package matchers
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/a8m/expect"
+)
+
+// ViaPolling matches by polling the child matcher until it returns a sucess.
+// It can take polling options test length and rate. Test length is the first
+// argument and it determines how long to poll. Rate is the second argument
+// and it sets what is the frequency to poll. ViaPolling will return success
+// the first time the child matcher returns a success. If the child matcher
+// never returns a nil, then it will return the last error.
+func ViaPolling(m expect.Matcher, pollingOpts ...string) expect.Matcher {
+	return pollingMatcher{
+		childMatcher: m,
+		pollingOpts:  pollingOpts,
+	}
+}
+
+// Always matches by polling the child matcher until it returns an error.
+// It can take polling options test length and rate. Test length is the first
+// argument and it determines how long to poll. Rate is the second argument
+// and it sets what is the frequency to poll. Always will return an error
+// the first time the child matcher returns an error. If the child matcher
+// never returns an error, then it will return a nil.
+func Always(m expect.Matcher, pollingOpts ...string) expect.Matcher {
```

Comment:
For both `Always` and `ViaPolling`, I'd like to see separate constructors instead of having a variadic argument for optional arguments.  We did the variadic-optional thing in [`stretchr/goweb`](https://github.com/stretchr/goweb/blob/master/map.go#L7-L86), and it succeeded in making the logic more complicated with very little benefit to our users.